### DISCUSSION TITLE
Azure DevOps macOS-13 VM .NET Maui build issue

### DISCUSSION THREAD

**Discussion by JohnHSE** [dotnet/maui#25052] [Q&A]
Hi all,
Just a heads up, Microsoft released an update to the macOS-13 VM over the weekend that breaks all .NET Maui builds. The only solution I have found so far is to use macOS-14 VM and switch to Xcode 16. However, .NET Maui does not yet support Xcode 16, so even if you can build it the apps are crashing.

Anyone from the Maui dev team able to light a fire under the Azure DevOps peeps to get this resolved? We are dead in the water...

https://developercommunity.visualstudio.com/t/iOS-MAUI-The-type-UIKitNSAdaptiveImage/10754377

Thanks
John

**Comment by DDHSchmidt** [comment]
Agree with the chaos this has caused but Maui aka the .Net iOS workload [*is* ready for Xcode 16](https://github.com/xamarin/xamarin-macios/releases/tag/dotnet-8.0.1xx-xcode16.0-8303) and that has been indeed the solution for us to just rush forward and use macos-14 vm together with Xcode 16.

Doing this has been easier for us than trying to pin down XCode versions, vmImage version, dotnet version, workload version, rollback files and what else...
It appears all the moving parts are way too hard to handle for the understaffed team (especially on the iOS side) and the modus operandi has been to embrace the cutting edge and just release fixes as quickly as possible.

I understand that and given the limited resources I even have to second that.
What I don't like on the other hand: There were [several](https://github.com/dotnet/maui/issues/25022) [regressions](https://github.com/xamarin/xamarin-macios/issues/20848) on [exploding iOS build time](https://github.com/xamarin/xamarin-macios/issues/21306) over the last months. Some people now brush this off as "[Surely it must have been your fault for choosing hosted vms and this is just hiccups you're observing](https://github.com/dotnet/maui/issues/25022#issuecomment-2385245451)"  I find this approach deceitful.

@rolfbjarne I'm still trying to establish a binlog for the last "fast" build, but researching all the versions used and rounding them up in a rollback file is time-consuming and customers are already at my door for another crash in our MAUI app, so don't hold your breath but rest assured I haven't forgotten it.
